lib/manager/poetry/readme.md

```diff
@@ -0,0 +1,170 @@
+
+pip
+
+## Package File Detection
+
+#### What type of package files and names does it use?
+
+_pyproject.toml_ and _poetry.lock_
+
+---
+
+#### What [fileMatch](https://renovatebot.com/docs/configuration-options/#filematch) pattern(s) should be used?
+
+_pyproject.toml_ and _poetry.lock_
+
+---
+
+#### Is it likely that many users would need to extend this pattern for custom file names?
+
+No
+
+---
+
+#### Is the fileMatch pattern likely to get many "false hits" for files that have nothing to do with package management?
+
+No
+
+## Parsing and Extraction
+
+#### Can package files have "local" links to each other that need to be resolved?
+
+Yes
+
+#### Is there reason why package files need to be parsed together (in serial) instead of independently?
+
+No
+
+---
+
+#### What format/syntax is the package file in? e.g. JSON, TOML, custom?
+
+TOML
+
+---
+
+#### How do you suggest parsing the file? Using an off-the-shelf parser, using regex, or can it be custom-parsed line by line?
+
+TOML can be parsed line by line
+
+---
+
+#### Does the package file structure distinguish between different "types" of dependencies? e.g. production dependencies, dev dependencies, etc?
+
+No
+
+---
+
+#### List all the sources/syntaxes of dependencies that can be extracted:
+
+TODO
+
+---
+
+#### Describe which types of dependencies above are supported and which will be implemented in future:
+
+## Versioning
+
+#### What versioning scheme do the package files use?
+
+<https://www.python.org/dev/peps/pep-0440>
+
+---
+
+#### Does this versioning scheme support range constraints, e.g. `^1.0.0` or `1.x`?
+
+Yes
+
+---
+
+#### Is this package manager used for applications, libraries, or both? If both, is there a way to tell which is which?
+
+Both, TODO
+
+---
+
+#### If ranges are supported, are there any cases when Renovate should pin ranges to exact versions if rangeStrategy=auto?
+
+It isn't clear yet
+
+## Lookup
+
+#### Is a new datasource required? Provide details
+
+No, it can reuse the pypi data source.
+
+---
+
+#### Will users need the capability to specify a custom host/registry to look up? Can it be found within the package files, or within other files inside the repository, or would it require Renovate configuration?
+
+Yes, TODO
+
+---
+
+#### Do the package files contain any "constraints" on the parent language (e.g. supports only v3.x of Python) or platform (Linux, Windows, etc) that should be used in the lookup procedure?
+
+Yes, language version can be defined using
+
+```toml
+[tool.poetry.dependencies]
+python = "^3.6"
+```
+
+platform TODO
+
+---
+
+#### Will users need the ability to configure language or other constraints using Renovate config?
+
+No
+
+## Artifacts
+
+#### Are lock files or checksum files used? Mandatory?
+
+Yes and yes, in _poetry.lock_
+
+---
+
+#### If so, what tool and exact commands should be used if updating 1 or more package versions in a dependency file?
+
+Update all dependencies
```

Comment:
About this, I think `update` command update the dependency according to the semver of the dependency inside pyproject.toml, and then update the lockfile

Reply:
What we want to be able to do here (ideally) is narrowly dependencies to specific versions.

For example let's say the version in the toml file is `1.0.0` but newer versions `1.0.1`, `1.1.0` and `2.0.0` exist. Depending on the user's config, they may even have those as 3 different PRs, so we need to way (ideally) to specifically update to that version.

Also, one branch/PR can have multiple packages grouped together, so we also ideally could update them all in one command (for efficiency). So something like `poetry update abc@1.0.1 def@2.2.0`.

Reply:
I have to try some stuff about this, but I don't think `poetry update` will update the pyproject.tom, I think it will only update lockfile.